### PR TITLE
Const get already imported

### PR DESCRIPTION
**Pull request by martong**
WIP

### CODE REVIEW

File: lib/AST/ExternalASTMerger.cpp [hunk 1]
```diff
@@ -144,14 +144,14 @@ class LazyASTImporter : public ASTImporter {
...
-      ToTag->setMustBuildLookupTable();
+      ToTag->getPrimaryContext()->setMustBuildLookupTable();
       assert(Parent.CanComplete(ToTag));
     } else if (auto *ToNamespace = dyn_cast<NamespaceDecl>(To)) {
       ToNamespace->setHasExternalVisibleStorage();
       assert(Parent.CanComplete(ToNamespace));
     } else if (auto *ToContainer = dyn_cast<ObjCContainerDecl>(To)) {
       ToContainer->setHasExternalLexicalStorage();
-      ToContainer->setMustBuildLookupTable();
+      ToContainer->getPrimaryContext()->setMustBuildLookupTable();
```
Comment (balazske) [inline lib/AST/ExternalASTMerger.cpp:154]: How are these changes related?

File: lib/AST/ASTImporter.cpp [hunk 2]
```diff
@@ -7174,12 +7177,10 @@ Attr *ASTImporter::Import(const Attr *FromAttr) {
   return ToAttr;
 }
 
-Decl *ASTImporter::GetAlreadyImportedOrNull(Decl *FromD) {
-  llvm::DenseMap<Decl *, Decl *>::iterator Pos = ImportedDecls.find(FromD);
+Decl *ASTImporter::GetAlreadyImportedOrNull(Decl *FromD) const {
```
Comment (balazske) [inline lib/AST/ASTImporter.cpp]: Probably the FromD can be const too?

File: lib/AST/ExternalASTMerger.cpp (hunk 1 above)
Comment (martong) [reply lib/AST/ExternalASTMerger.cpp:154]: By removing the `ImportDefinitionIfNeeded` from `GetAlreadyImportedOrNull` a new assertion is triggered here. `setMustBuildLookupTable` must alwasy be called on a primary context.

File: lib/AST/ASTImporter.cpp (hunk 2 above)
Comment (martong) [reply lib/AST/ASTImporter.cpp]: :+1: Good point.

**Review by balazske — APPROVED** [review]

**Comment by martong** [comment]
Update: squashed commits.

**Comment by martong** [comment]
https://reviews.llvm.org/D53755
